rllib/core/models/tf/heads.py

```diff
@@ -18,6 +20,81 @@
 tf1, tf, tfv = try_import_tf()
 
 
+def auto_fold_unfold_time(input_spec: str):
+    """Automatically folds/unfolds the time dimension of a tensor.
+
+    This is useful when calling the model requires a batch dimension only, but the
+    input data has a batch- and a time-dimension. This decorator will automatically
+    fold the time dimension into the batch dimension before calling the model and
+    unfold the batch dimension back into the time dimension after calling the model.
+
+    Args:
+        input_spec: The input spec of the model.
+
+    Returns:
+        A decorator that automatically folds/unfolds the time_dimension if present.
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, input_data, **kwargs):
+            if not hasattr(self, input_spec):
+                raise ValueError(
+                    "The model must have an input_specs attribute to "
+                    "automatically fold/unfold the time dimension."
+                )
+            if not tf.is_tensor(input_data):
+                raise ValueError(
+                    f"input_data must be a tf.Tensor to fold/unfold "
+                    f"time automatically, but got {type(input_data)}."
+                )
+            # Attempt to fold/unfold the time dimension.
+            actual_shape = tf.shape(input_data)
+            spec = getattr(self, input_spec)
+
+            try:
+                # Validate the input data against the input spec to find out it we
+                # should attempt to fold/unfold the time dimension.
+                spec.validate(input_data)
```

Comment:
This is an unsolved issue with this PR. We need to be able to get around spec-checking in the mid-term.